### ISSUE TITLE
[FR] Visual Sync Indicator and Background Synchronization for Android

### ISSUE THREAD

**Issue by MarioSertorio** [AppFlowy-IO/AppFlowy#6192]
### Description

I suggest adding a visual indicator in the AppFlowy Android app to show when content is being synchronized with the remote server. This would provide users with clear feedback that their changes are being saved and updated. Additionally, enabling the synchronization process to run in the background would improve app usability and performance by allowing users to continue working without interruption.

### Impact

All AppFlowy Android users would benefit from these enhancements, particularly those who frequently use the app on the go and rely on real-time data synchronization to keep their notes and content up to date across multiple devices.

### Additional Context

These changes would help create a more seamless user experience and prevent confusion or frustration from lack of feedback during the sync process. Background synchronization is a common feature in many productivity apps and would align AppFlowy with current user expectations.

**Comment by capybaras-team** [comment]
Yes! We definitely need at least an indicator for successful document synchronization, and not just on Android, but also on PC. There should also be a loader showing the synchronization process itself. It would add +1000% to the perception of the app as reliable.

**Comment by andrewragy** [comment]
Any progress on the sync indicator? Is it still planned?
